qiskit/transpiler/passes/scheduling/base_scheduler.py

```diff
@@ -261,10 +261,21 @@ def _get_node_duration(
     ) -> int:
         """A helper method to get duration from node or calibration."""
         indices = [dag.find_bit(qarg).index for qarg in node.qargs]
+        is_mid_circuit = any(
+            dag.descendants(successor) != set() for successor in dag.successors(node)
+        )
 
         if dag.has_calibration_for(node):
             # If node has calibration, this value should be the highest priority
             cal_key = tuple(indices), tuple(float(p) for p in node.op.params)
+            if isinstance(node.op, Reset):
+                raise UserWarning(
```

Comment:
I think a `RuntimeWarning` is probably more appropriate here, it's documented in the python docs as being used for "warnings about dubious runtime features": https://docs.python.org/3/library/warnings.html#warning-categories